src/smp/ipi.c

```diff
@@ -12,6 +12,28 @@
 #include <smp/ipi.h>
 #include <smp/lock.h>
 
+static struct ipi_args ipi_args = { 0 };
+
+static inline void init_ipi_args(IpiRemoteCall_t func, word_t data0,
+                                 word_t data1, word_t data2, word_t mask)
+{
+    ipi_args.remoteCall = (word_t)func;
+    ipi_args.args[0] = data0;
+    ipi_args.args[1] = data1;
+    ipi_args.args[2] = data2;
+
+    /* Get number of cores involved in this IPI */
+    ipi_args.totalCoreBarrier = popcountl(mask);
+
+#ifdef CONFIG_ARCH_ARM
+    /* This is to make sure if a write to ebig_kernel_lock.node_owners[target_core].ipi
+     * is observed, the above writes are also observed by readers according the
+     * address dependancy.
+     */
+    asm volatile("dmb ishst" ::: "memory");
```

Comment:
You could use `IPI_MEM_BARRIER` here. That's a dsb instead of a dmb, so perhaps stronger (and slightly slower) than needed.